src/datastore/helpers.ts

```diff
@@ -898,3 +898,18 @@ export function registerMempoolPromStats(pgEvents: PgStoreEventEmitter) {
     });
   });
 }
+
+export function convertTxQueryResultToDbMempoolTx(txs: TxQueryResult[]): DbMempoolTx[] {
+  const dbMempoolTxs: DbMempoolTx[] = [];
+  for (const tx of txs) {
+    const dbMempoolTx: DbMempoolTx = Object.assign(tx, {
+      pruned: false,
+      receipt_time: Math.round(Date.now() / 1000),
+      fee_rate: BigInt(tx.fee_rate),
+      token_transfer_amount: BigInt(tx.token_transfer_amount ?? 0),
```

Comment:
It doesn't seem like this function fulfills the "convert mined tx to mempool tx" correctly, or it at least it has some confusing lines. For example, isn't the difference between the two object types only the `receipt_time` and `pruned` properties?

I see the line `token_transfer_amount` which is specific only to `token_transfer` tx types -- why aren't the other tx-type-specific properties also here? It looks like perhaps that is just an unnecessary property? If it is necessary, something seems wrong, because isn't `token_transfer_amount` supposed to be null/undefined for txs that are not of type `token_transfer`?

Reply:
These other properties are here because the types TxQueryResult and DbMempoolTx do not match up 1:1. There are type differences that I am accounting for in addition to adding receipt_time and pruned properties.

Your're right about token_transfer_amount. I was converting the type incorrectly here.